tests/e2e/testdata/dag-inline.yaml

```
@@ -0,0 +1,28 @@
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: dag-inline-
+spec:
+  entrypoint: main
+  templates:
+    - name: main
+      dag:
+        tasks:
+          - name: task-0
+            spec:
+              container:
+                image: argoproj/argosay:v2
+          - name: task-1
+            dependencies:
+              - task-0
+            arguments:
+              parameters:
+                - name: x
+                  value: "{{tasks.task-0.outputs.result}}"
+            spec:
```

I think `spec` could be ambiguous here since it could imply the spec of the `task`, not a template. Maybe

```suggestion
            inlineTemplate:
```
?

---

the best name would be `template` obviously, but that is taken!